### ISSUE TITLE
Environment crashes because it seems to be overriding built in modules

### ISSUE THREAD

**Issue by mtrattner** [horovod/horovod#4042]
**Environment:**
Here my installed packages (Python 3.10.8):
absl-py==2.1.0
appnope==0.1.4
astropy==6.1.0
astropy-iers-data==0.2024.5.6.0.29.28
asttokens==2.4.1
astunparse==1.6.3
cachetools==5.3.3
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
cloudpickle==3.0.0
comm==0.2.2
contourpy==1.2.1
cycler==0.12.1
debugpy==1.8.1
decorator==5.1.1
exceptiongroup==1.2.1
executing==2.0.1
flatbuffers==24.3.25
fonttools==4.51.0
gast==0.4.0
google-auth==2.29.0
google-auth-oauthlib==1.0.0
google-pasta==0.2.0
grpcio==1.63.0
h5py==3.11.0
horovod==0.28.1
idna==3.7
imageio==2.34.1
ipykernel==6.29.4
ipython==8.24.0
jedi==0.19.1
jupyter_client==8.6.1
jupyter_core==5.7.2
keras==2.13.1
kiwisolver==1.4.5
lazy_loader==0.4
libclang==18.1.1
llvmlite==0.42.0
Markdown==3.6
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.8.4
matplotlib-inline==0.1.7
mdurl==0.1.2
nest-asyncio==1.6.0
networkx==3.3
numba==0.59.1
numpy==1.24.3
oauthlib==3.2.2
opt-einsum==3.3.0
packaging==24.0
pandas==2.2.2
parso==0.8.4
pexpect==4.9.0
pillow==10.3.0
platformdirs==4.2.1
prompt-toolkit==3.0.43
protobuf==4.25.3
psutil==5.9.8
ptyprocess==0.7.0
pure-eval==0.2.2
pyasn1==0.6.0
pyasn1_modules==0.4.0
pycparser==2.22
pyerfa==2.0.1.4
Pygments==2.18.0
pyparsing==3.1.2
python-dateutil==2.9.0.post0
pytz==2024.1
PyYAML==6.0.1
pyzmq==26.0.3
requests==2.31.0
requests-oauthlib==2.0.0
rich==13.7.1
rsa==4.9
scikit-image==0.23.2
scipy==1.13.0
six==1.16.0
stack-data==0.6.3
tensorboard==2.13.0
tensorboard-data-server==0.7.2
tensorflow==2.13.0
tensorflow-estimator==2.13.0
tensorflow-io-gcs-filesystem==0.37.0
termcolor==2.4.0
tifffile==2024.5.3
tornado==6.4
tqdm==4.66.1
traitlets==5.14.3
typing_extensions==4.5.0
tzdata==2024.1
urllib3==2.2.1
wcwidth==0.2.13
Werkzeug==3.0.3
wrapt==1.16.0
your==0.6.7

**Bug report:**
When installing Horovod and Tensorflow with pip I am getting the following error when trying to run any python code in this environment:

> The file 'venv/lib/python3.10/site-packages/typing_extensions.py' seems to be overriding built in modules and interfering with the startup of the kernel. Consider renaming the file and starting the kernel again.

The issue does not appear when installing Horovod or Tensorflow independently. Some dependencies don't seem to agree with each other so I provided my output from pip freeze above for you to be able to reproduce the issue.
